appscale/tools/appscale_tools.py

```diff
@@ -604,28 +604,32 @@ def terminate_instances(cls, options):
       AppScaleException: If AppScale is not running, and thus can't be
       terminated.
     """
-    if not os.path.exists(LocalState.get_secret_key_location(options.keyname)):
-      raise AppScaleException("AppScale is not running with the keyname {0}".
+    try:
+      infrastructure = LocalState.get_infrastructure(options.keyname)
+    except IOError:
+      raise AppScaleException("Cannot find AppScale's configuration for keyname {0}".
         format(options.keyname))
 
-    infrastructure = LocalState.get_infrastructure(options.keyname)
-
-    # If the user is on a cloud deployment, and not backing their data to
-    # persistent disks, warn them before shutting down AppScale.
-    # Also, if we're in developer mode, skip the warning.
-    if infrastructure != "xen" and not LocalState.are_disks_used(
-      options.keyname) and not options.test:
-      LocalState.ensure_user_wants_to_terminate()
+    if infrastructure == "xen" or not options.terminate:
+      # We are in cluster mode: let's chek AppScale is running.
+      if not os.path.exists(LocalState.get_secret_key_location(options.keyname)):
+        raise AppScaleException("AppScale is not running with the keyname {0}".
+          format(options.keyname))
 
-    if infrastructure in InfrastructureAgentFactory.VALID_AGENTS:
-      RemoteHelper.terminate_cloud_infrastructure(options.keyname,
-        options.verbose)
-    else:
+    # Stop gracefully the AppScale deployment.
+    try:
       RemoteHelper.terminate_virtualized_cluster(options.keyname,
         options.verbose)
+    except:
+      # Don't fail if we cannot find the configuration.
+      pass
 
-    LocalState.cleanup_appscale_files(options.keyname)
-    AppScaleLogger.success("Successfully shut down your AppScale deployment.")
```

Comment:
What's the output on a cluster? Does the user get any feedback on whether their `down` completed successfully?